gemrb/core/Projectile.h

```diff
@@ -198,6 +188,18 @@ struct ProjectileExtension
 
 class GEM_EXPORT Projectile
 {
+private:
+	enum class ProjectileState {
+		NEW,
+		TRAVELLING,       // moving towards target
+		AWAITING_HIT,     // arrived
+		HIT,              // hit
```

Comment:
These name and comment combinations are confusing. I propose we use AWAITING_TRIGGER instead, since it's only relevant for projectiles with triggers (this is the same nomenclature other tools use). "hit" and "arrived" are true for both values, so I propose to change the trigger comment to "trap set".

Reply:
Sorry, maybe I wasn't clear enough, but it's AWAITING_HIT that is for the skulltrap behaviour (was P_TRIGGER before), not the other way around.

Reply:
Please write an edit suggestion for these lines, so I'm not left confused.

Reply:
AWAITING_HIT,     // arrived
		HIT,              // hit

to

		AWAITING_TRIGGER,     // hovering over target, waiting to trigger
		HIT,              // just hit target